packages/host/tests/acceptance/code-submode/create-file-test.gts

```diff
@@ -467,4 +461,215 @@ module('Acceptance | code submode | create-file tests', function (hooks) {
 
     await deferred.promise;
   });
+
+  test<TestContextWithSave>('can create a new card definition', async function (assert) {
+    assert.expect(7);
+    let expectedSrc = `
+import { CardDef } from 'https://cardstack.com/base/card-api';
+export class TestCard extends CardDef {
+  static displayName = "Test Card";
+}`;
+    await openNewFileModal('Card Definition');
+    assert
+      .dom('[data-test-create-definition]')
+      .isDisabled('create button is disabled');
+    await fillIn('[data-test-display-name-field]', 'Test Card');
+    assert
+      .dom('[data-test-create-definition]')
+      .isDisabled('create button is disabled');
+    await fillIn('[data-test-file-name-field]', 'test-card');
+    assert
+      .dom('[data-test-create-definition]')
+      .isEnabled('create button is enabled');
+
+    await percySnapshot(assert);
+
+    this.onSave((content) => {
+      if (typeof content !== 'string') {
+        throw new Error(`expected string save data`);
+      }
+      assert.strictEqual(content, expectedSrc, 'the source is correct');
+    });
+    await click('[data-test-create-definition]');
+    await waitFor('[data-test-create-file-modal]', { count: 0 });
+    await waitForCodeEditor();
+    assert.strictEqual(
+      getMonacoContent(),
+      expectedSrc,
+      'monaco displays the new definition',
+    );
+
+    await waitFor('[data-test-card-schema="Test Card"]');
+    assert
+      .dom('[data-test-card-schema]')
+      .exists({ count: 3 }, 'the card hierarchy is displayed in schema editor');
+    assert.dom('[data-test-total-fields]').containsText('3 Fields');
+  });
+
+  skip<TestContextWithSave>('TODO can create a new field definition', async function (_assert) {
+    // TODO include percy snapshot
+  });
```

Comment:
Is this for a future PR?

Reply:
Yeah, I have a ticket for new field definition opened. Luckily that will be very easy to do after this PR 😁